ci: generates release notes with helm chart updates workflow

JBAhire opened the pull request:
## Description
This PR,
- modifies script to update helm charts to facilitate release notes generation. 
- We are using `github-changelog` python library to get release notes for all the services and creates `RELEASE_NOTES.md` file. 
- Maintain compatibility matrix file so we can get a changelog from last version to latest. 


### Testing
I have tested this workflow on my personal fork and it works as expected. 

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### Documentation

kotharironak commented on .github/workflows/create-charts.sh:
```diff
@@ -75,14 +75,54 @@ update_platform_services_charts() {
+}
+
+write_changelog() {
+    echo '### Attribute service ('$(cat "kubernetes/tmp/compatibility_matrix.txt" | grep "^attribute-service:" | cut -d":" -f2-) '--->' $attribute_service_version')'
```
Can we do this as below?

```
attribute_service_piror_version=get_prior_service_version('attribute-service')
    echo '### Attribute service ('$attribute_service_piror_version '--->' $attribute_service_version')'
    changelog -m hypertrace attribute-service $attribute_service_piror_version $attribute_service_version
```

and having a common function to get prior version

```
get_prior_service_version(service_name) {
  return $(cat "kubernetes/tmp/compatibility_matrix.txt" | grep "^${service_name}:" | cut -d":" -f2-)
}
```

kotharironak commented on .github/workflows/create-charts.sh:
```diff
@@ -75,14 +75,54 @@ update_platform_services_charts() {
+    write_changelog > "${release_notes_file}"
+fi
+
+if [ $1 == "compatibility-matrix" ]; then
```
Can we have some documentation that how this works here?

kotharironak reviewed (commented):
Can we also add the link of release notes into slack notification?

kotharironak commented on .github/workflows/create-charts.sh:
```diff
@@ -75,14 +75,54 @@ update_platform_services_charts() {
+write_changelog() {
+    echo '### Attribute service ('$(cat "kubernetes/tmp/compatibility_matrix.txt" | grep "^attribute-service:" | cut -d":" -f2-) '--->' $attribute_service_version')'
+    changelog -m hypertrace attribute-service $(cat "kubernetes/tmp/compatibility_matrix.txt" | grep "^attribute-service:" | cut -d":" -f2-) $attribute_service_version
+    echo '### Gateway service ('$(cat "kubernetes/tmp/compatibility_matrix.txt" | grep "^gateway-service:" | cut -d":" -f2-) '--->' $gateway_service_version')'
```
Instead of `--->`, can we follow `0.1.53 ... 0.1.54`?

JBAhire commented:
@kotharironak addressed your comments and updated the formatting of release notes. You can view updates notes here: https://github.com/hypertrace/hypertrace/releases/tag/0.1.4

rish691 commented:
> @kotharironak addressed your comments and updated the formatting of release notes. You can view updates notes here: https://github.com/hypertrace/hypertrace/releases/tag/0.1.4

@JBAhire should have three dots here `Attribute service (0.9.3 .. 0.9.3)` ?

rish691 commented on RELEASE_NOTES.md:
```diff
@@ -0,0 +1,21 @@
+#### `Attribute service`  (0.9.3 .. 0.9.3)
```
Why are we checking in this file?

JBAhire replied:
Currently, as we are releasing manually, we can copy paste them. But we will be moving it to automated release once we have self-hosted runner for e2e tests in place. So that time, it will just `cat` this file and paste it in release description. :)
